src/convertToObject.js

```diff
@@ -6,7 +6,20 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
-}
+  const sourcesArr = sourceString.split(';');
+
+  const resultObj = sourcesArr.reduce((acumulator, item) => {
```

Comment:
[CODE KNOWLEDGE] - Don't need to make a copy before each operation. Moreover, methods like `filter` or `map` don't mutate the original array. In this case, you've used `reduce` correctly as it transforms an array into an object without mutating the original array.